pkg/controller/volume/persistentvolume/controller_base.go

```diff
@@ -422,12 +423,15 @@ func (ctrl *PersistentVolumeController) deleteClaim(obj interface{}) {
 func (ctrl *PersistentVolumeController) Run() {
 	glog.V(4).Infof("starting PersistentVolumeController")
 
-	ctrl.initializeCaches(ctrl.volumeSource, ctrl.claimSource)
+	ctrl.initializeCaches(ctrl.volumes.store, ctrl.claimSource)
```

Comment:
`initializeCaches` should copy all volumes from the shared informer into internal `ctrl.volumes`.